src/main/java/net/neoforged/neoforge/items/ContainerOrHandler.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.items;
+
+import net.minecraft.world.Container;
+import org.jetbrains.annotations.Nullable;
+
+public record ContainerOrHandler(
+        @Nullable Container container,
+        @Nullable IItemHandler itemHandler) {
```

Comment:
Should we enforce through the canonical constructor that either of the two components but not both must have a value? That's the implication of the name, at least.

Reply:
Yes, why not!